Adding aesgcm info back in and mentioning migrating between types (#59554)
openshift/openshift-docs: modules/enabling-etcd-encryption.adoc

* A disk I/O can affect the node that receives the backup state.
====

You can encrypt the etcd database in either AES-GCM or AES-CBC encryption.

[NOTE]
====

> To migrate your etcd database from one encryption type to the other, you can modify the API server's `spec.encryption.type` field. {product-title} then migrates the encryption type without having to unencrypt the etcd database first.

Reviewer: I'm not sure what the `{product-title}` will be? I would like to change this sentence to be "Then data encryption migration to the new encryption type will happen automatically" or something along those lines.

Author: Hey @dinhxuanvu if you look at the preview link I shared (https://59554--docspreview.netlify.app/openshift-enterprise/latest/security/encrypting-etcd.html#enabling-etcd-encryption_encrypting-etcd) you can see how it resolves. {product-title} will turn into "OpenShift Container Platform".

However, I like your suggestion, so I've reworded similar to that. PTAL now and let me know what you think. Thanks!

Note that the auto previews actually just stopped working right now, so I'm uploading a new preview here (VPN required): 

http://file.rdu.redhat.com/~ahoffer/2023/OSDOCS-5977/security/encrypting-etcd.html#enabling-etcd-encryption_encrypting-etcd

> If you want to migrate your etcd database from one encryption type to the other, you can modify the API server's `spec.encryption.type` field, and the etcd Operator will migrate the encryption type without having to unencrypt the etcd database first.

Reviewer 2: suggestion:
```suggestion
To migrate your etcd database from one encryption type to the other, you can modify the API server's `spec.encryption.type` field. The etcd Operator then migrates the encryption type without having to unencrypt the etcd database first.
```
